### PR TITLE
Key api

### LINKED ISSUE #13: Implement APIs around `keys` schema entity
## Description

Implement the APIs around the `keys` DB entity. 
Make sure your PR has tests and that you are attaching relevant, postman collections, cURL scripts and screen recordings are required along with the PR.

## References
- [FusionAuth Keys API Docs](https://fusionauth.io/docs/apis/keys)
- [DB Schema](https://dbdiagram.io/d/OIDC-Wrapper-DB-Schema-665d7a3bb65d933879567dd2)

### PR DESCRIPTION
**Pull request by Zolo-Ryan**
It contains key api routes as well as group api part 
group api testing postman collection : [collection](https://www.postman.com/planetary-water-899223/workspace/ashu-s-public-ws/collection/29097578-a608735e-6c13-4569-8615-fd024e6c4461?action=share&creator=29097578)
folders to be reviewed : /src/group

key api testing postman collection [collection](https://www.postman.com/planetary-water-899223/workspace/ashu-s-public-ws/collection/29097578-95fd372c-ce63-4095-b8a9-b044fef8fdaa?action=share&creator=29097578)
folders to reviewed : /src/key-api

### CODE REVIEW

File: src/key/key.service.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    constructor(private readonly prismaService : PrismaService){
+        this.logger = new Logger() ;
+    }
+
+    async retrieveAllKey(){
+        try{
+            const item = await this.prismaService.key.findMany();
+            if(!item){
+                return {
+                    success : true,
```
Comment (BackendBananeWala) [inline src/key/key.service.ts]: success : false?

File: src/key/key.service.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        const id = uuid 
+        try{
+            const item = await this.prismaService.key.findUnique({ where : {id}})
+            if(!item){
+                throw new HttpException(
+                    'key does not exist with provided id',
+                    HttpStatus.NOT_FOUND
+                );
+            }
+            return {
```
Comment (BackendBananeWala) [inline src/key/key.service.ts:52]: success: true
item should be inside data key

File: src/key/key.service.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                    key.id = uuid;
+                    key.issuer = issuer;
+                    key.length = length ;
+                    key.userName = name ;
+                    key.type = 'RSA'
+                  });
+                return {
+                    success : true,
+                    message : 'key generated successfully',
+                    jwks
```
Comment (BackendBananeWala) [inline src/key/key.service.ts]: inside data key

File: src/key/key.service.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
```
Comment (BackendBananeWala) [inline src/key/key.service.ts]: inside data key

File: src/key/key.service.ts [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                    key.insertinstant = insertinstant ;
+                    key.id = uuid;
+                    key.userName = name ;
+                    key.kid = kid;
+                    key.type = 'HMAC'
+                  });
+                return {
+                    success : true,
+                    message : 'key generated successfully',
+                    jwks
```
Comment (BackendBananeWala) [inline src/key/key.service.ts]: inside data key

**Comment by Ashu463** [comment]
Hey @BackendBananeWala all changes are updated along with that generate key api part added

**Pull request closed without merging**